feat(test-utils): add renderHook wrapper (#2144)
commercetools/merchant-center-application-kit: .changeset/heavy-ligers-admire.md

---
'@commercetools-frontend/application-shell': minor
---

> Adds re-export of `@testing-library/react-hooks` from `test-utils` with `renderHook` functions wrapped with app-kit providers.

Reviewer: ```suggestion
The `@commercetools-frontend/application-shell/test-utils` now exports a new utility function for testing hooks: `renderHook`. This function wraps the original `renderHook` function from `@testing-library/react-hooks` but it comes pre-configured with providers for testing Custom Applications.
```

> The `renderHook` function is wrapped similarily to the existing `renderAppWithRedux`:
> 
> - All `options` which can be passed to `renderAppWithRedux` can be passed to `renderHooks`
> - All additional poperties returned as a result of an `renderAppWithRedux` call (like `store` or `history`) are returend from the wrapped `renderHook` too

Reviewer: ```suggestion
The new `renderHook` function also accepts the same options as other test utils, such as `renderApp`, and returns some useful properties like `store`, `history`, etc.
```

> All `@testing-library/react-hooks` functionality is exported under the namespace `hooks` from the package '@commercetools-frontend/application-shell/test-utils'.

Reviewer: ```suggestion
```

---
'@commercetools-frontend/application-shell': minor
---

> feat(test-utils): add wrapped renderHook util

Reviewer 2: a small reference, here is how I did it in another PR

'@commercetools-frontend/application-shell': minor
---

Re-export `@testing-library/react-hooks` from `test-utils` with `renderHook` functions wrapped in all the app-kit providers.

`renderHooks` is wrapped the same way the existing `renderAppWithRedux` helper is:

- all the `options` which can be passed to `renderAppWithRedux`can be passed to `renderHooks` as well
- all the additional poperties returned as a result of `renderAppWithRedux` invokation (like `store` or `history`) is returend from the wrapped `renderHooks` too.

All the `@testing-library/react-hooks` content is exported under the namespace `hooks` from the package '@commercetools-frontend/application-shell/test-utils'.

**Usage example**

```jsx
import { hooks } from '@commercetools-frontend/application-shell/test-utils';

const { act, renderHook } = hooks;

it('should navigate to a route', () => {
  const { result, history } = renderHook(useRoutes);
  act(() => result.current.someRoute.go());
  expect(history.location.pathname).toBe('/some-route');
});

> ```

Reviewer 2: yeaaah boooooi 😍
